### PR TITLE
Dalvik Bytecode deocding

### PR DESCRIPTION
**Pull request by gnieto**
I've implemented most of the Dalvik operations (there are some instructions missing: _binop/lit16_, _ binop/lit8_, _invoke-polymorphic_, _invoke-polymorphic/range_, _invoke-custom_, _invoke-custom/range_).

I've created a `ByteCodeDecoder` which is an iterator that yeilds `ByteCode` from the source.

This is not used anywhere else (yet), as the code segments are not yet parsed on develop.

### CODE REVIEW

**Review by Razican — CHANGES_REQUESTED** [review]
Incredible work!! Thanks!

Seems that tests are failing due to formatting. Could you check it? Also, seems we have a lot of warnings from missing docs. Maybe it's not so important right now, but it will be in the future. Let me know what you think.

**Comment by gnieto** [comment]
I've added the allow missing docs on the bytecodes structures/enums as it will not add any value at the moment (if you are using this, probably you are also looking into dalvik documentation, as it will be a nicer source of data).

Besides this, formatting is ok now and I've added some new documentation.

File: src/lib.rs [hunk 1]
```diff
@@ -103,7 +104,7 @@ impl Dex {
 }
 
 impl From<DexReader> for Dex {
-    fn from(reader: DexReader) -> Dex {
+    fn from(_: DexReader) -> Dex {
```
Comment (Razican) [inline src/lib.rs]: Actually, I like this warning, since it remembers what it's still not implemented.

**Comment by gnieto** [comment]
Added missing instructions

**Review by Razican — APPROVED** [review]

**Comment by Razican** [comment]
Amazing work!!! 🎉

**Pull request merged**